plotly.agent.nut

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2015 Electric Imp
+// This file is licensed under the MIT License
+// http://opensource.org/licenses/MIT
+
+class Plotly {
+    
+    // Use this function to get a timestamp that Plotly will automatically 
+    // recognize and style correctly.
+    static function getPlotlyTimestamp() {
+        local timestamp = date();
+        return format("%04i-%02i-%02i %02i:%02i:%02i",
+            timestamp.year, timestamp.month, timestamp.day,
+            timestamp.hour, timestamp.min, timestamp.sec);
+    }
+    
+    static PLOTLY_ENDPOINT = "https://plot.ly/clientresp";
+    static PLOTLY_PLATFORM = "electricimp"
+    
+    static PlotlyMessageType = {
+        plot = "plot",
+        style = "style",
+        layout = "layout"
+    }
+
+    _url = null;
+    _username = null;
+    _userKey = null;
+    _filename = null;
+    _worldReadable = null;
+    _persistentLayout = null;
+    _persistentStyle = null;
+    _hack_sent_first_plot = null;
+
+    
+    function constructor(username, userKey,
+                         filename, worldReadable, traces, 
+                         callback = null) {
+        _url = "";
+        _username = username;
+        _userKey = userKey;
+        _filename = filename;
+        _worldReadable = worldReadable;
+        _persistentLayout = {"xaxis" : {}, "yaxis" : {}};
+        _persistentStyle = [];
+        local plotlyInput = [];
+        foreach(trace in traces) {
+            plotlyInput.append({
+                "x" : [],
+                "y" : [],
+                "name" : trace
+            });
+            _persistentStyle.append({
+                "name" : trace  
+            });
+        };
+        _hack_sent_first_plot = false; 
```

Comment:
Having _hack in a property name generally implies you should be cleaning up some code, or renaming a property..